packages/core/src/lib/position.mts

```diff
@@ -0,0 +1,49 @@
+type Change<T> = {
+  previous: T;
+  current: T;
+};
+
+type PageInfo = {
+  active: Change<number>;
+  total: number;
+  pageSize: number;
+};
+
+/**
+ * Given information about a page, decides the next position at which the active
+ * item should be rendered in the page.
+ */
+type PositionReducer = (info: PageInfo) => (pointer: number) => number;
+
+/**
+ * Creates the next position for the active item considering a finite list of
+ * items to be rendered on a page.
+ */
+export const finite: PositionReducer =
+  ({ pageSize, total, active }) =>
+  () => {
+    const middle = Math.floor(pageSize / 2);
+    return total <= pageSize || active.current < middle
+      ? active.current
+      : active.current >= total - middle
+      ? active.current + pageSize - total
+      : middle;
+  };
```

Comment:
For example,
for page size `5`
and items
`[A, B, C, D, E, F, G, H, I , J]`
will create corresponding positions
`[0, 1, 2, 2, 2, 2, 2, 2, 3, 4]`